File: Single-Agent/research-agent/frontend/src/types/research.ts
export interface Reference {
  website_name: string;
  article_title: string;
  url: string;
  published_date?: string;
  author?: string;
  credibility_score?: number;
}

export interface ResearchFilters {
  date_range?: string;
  sources?: string[];
  min_confidence?: number;
  category?: string;
}

export interface ResearchResponse {
  status: string;
  agent: string;
  request_id: string;
  timestamp: string;
  confidence: number;
  summary: string;
  executive_summary?: string;
  keywords: string[];
  references: Reference[];
  recommendations: string[];
  execution_time: string;
  fact_check_details?: {
    verified: boolean;
    confidence_level: string;
    matching_source_count: number;
    contradictions_count: number;
    verification_status: string;
    details: any[];
  };
}

export interface HistoryItem {
  id: string;
  objective: string;
  confidence: number;
  summary: string;
  created_at: string;
  execution_time: string;
  source_count: number;
}

export interface AnalyticsMetrics {
  total_requests: number;
  todays_searches: number;
  sources_used: number;
  average_confidence: number;
  average_response_time: string;
}

export interface AnalyticsData {
  metrics: AnalyticsMetrics;
  charts: {
    source_distribution: { name: string; value: number; color: string }[];
    confidence_distribution: { range: string; count: number }[];
    top_topics: { topic: string; count: number }[];
    most_used_sources: { domain: string; count: number }[];
  };
}

export interface User {
  id: string;
  email: string;
  full_name?: string;
}

File: Single-Agent/research-agent/backend/schemas/research.py
from pydantic import BaseModel, Field, HttpUrl
from typing import List, Optional, Dict, Any
from uuid import UUID
from datetime import datetime

class ResearchFilterSchema(BaseModel):
    date_range: Optional[str] = "all"
    sources: Optional[List[str]] = Field(default_factory=lambda: ["web", "docs", "github", "papers"])
    min_confidence: Optional[int] = 0
    category: Optional[str] = "General"

class ResearchStartRequest(BaseModel):
    objective: str = Field(..., description="The user's research goal or question")
    filters: Optional[ResearchFilterSchema] = Field(default_factory=ResearchFilterSchema)
    max_results: Optional[int] = 8

class QuickSearchRequest(BaseModel):
    query: str
    search_type: Optional[str] = "all"

class SummarizeRequest(BaseModel):
    content: str
    url: Optional[str] = None
    target_length: Optional[str] = "medium"

class CompareRequest(BaseModel):
    topics: List[str] = Field(..., min_items=2)
    aspects: Optional[List[str]] = None

class FactCheckRequest(BaseModel):
    claim: str
    context: Optional[str] = None

class ReferenceSchema(BaseModel):
    website_name: str
    article_title: str
    url: str
    published_date: Optional[str] = "N/A"
    author: Optional[str] = "N/A"
    credibility_score: Optional[float] = 0.85

class LifeOSResearchResponse(BaseModel):
    status: str = "success"
    agent: str = "Research"
    request_id: str
    timestamp: str
    confidence: int = Field(..., ge=0, le=100)
    summary: str
    executive_summary: Optional[str] = None
    keywords: List[str] = Field(default_factory=list)
    references: List[ReferenceSchema] = Field(default_factory=list)
    recommendations: List[str] = Field(default_factory=list)
    execution_time: str
    fact_check_details: Optional[Dict[str, Any]] = None

class ResearchHistoryItem(BaseModel):
    id: UUID
    objective: str
    confidence: int
    summary: str
    created_at: str
    execution_time: str
    source_count: int

class ResearchHistoryResponse(BaseModel):
    status: str = "success"
    total: int
    items: List[ResearchHistoryItem]
